charm/src/charm.ts

```diff
@@ -319,6 +316,242 @@ export class CharmManager {
+   */
+  getReadingFrom(charm: Cell<Charm>): Cell<Charm>[] {
+    // Get all charms that might be referenced
+    const allCharms = this.getCharms().get();
+    const result: Cell<Charm>[] = [];
+    const seenEntityIds = new Set<string>(); // Track entities we've already processed
+
+    if (!charm) return result;
+
+    try {
+      // Get the argument data - this is where references to other charms are stored
+      const argumentCell = this.getArgument(charm);
+      if (!argumentCell) return result;
+
+      // Get the raw argument value
+      const argumentLink = argumentCell.getAsCellLink();
+      if (!argumentLink || !argumentLink.cell) return result;
+
+      const argumentValue = argumentLink.cell.getAtPath(argumentLink.path);
+
+      // Helper function to add a matching charm to the result
+      const addMatchingCharm = (docId: EntityId) => {
+        if (!docId || !docId["/"]) return;
+
+        const entityIdStr = typeof docId["/"] === "string"
+          ? docId["/"]
+          : JSON.stringify(docId["/"]);
+
+        // Skip if we've already processed this entity
+        if (seenEntityIds.has(entityIdStr)) return;
+        seenEntityIds.add(entityIdStr);
+
+        // Find matching charm by entity ID
+        const matchingCharm = allCharms.find((c) => {
+          const cId = getEntityId(c);
+          return cId && docId["/"] === cId["/"];
+        });
+
+        if (
+          matchingCharm && !isSameEntity(matchingCharm, charm) &&
+          !result.some((c) => isSameEntity(c, matchingCharm))
+        ) {
+          result.push(matchingCharm);
+          // Reference added to result
+        }
+      };
+
+      // SIMPLEST APPROACH: Only scan the top-level properties for references
+      // This avoids any recursion and potential stack overflow issues
+      // Scan top-level properties for references
+      if (argumentValue && typeof argumentValue === "object") {
+        // Check each top-level property
+        for (const key in argumentValue) {
+          const value = argumentValue[key];
+          if (!value || typeof value !== "object") continue;
+
+          // Look for alias references ($alias property)
+          if (value.$alias && value.$alias.cell) {
+            const aliasId = getEntityId(value.$alias.cell);
+            if (aliasId && aliasId["/"]) {
+              // Found alias reference
+              addMatchingCharm(aliasId);
+            }
+          }
+
+          // Look for direct cell references (cell + path properties)
+          if (value.cell && value.path !== undefined) {
+            const cellId = getEntityId(value.cell);
+            if (cellId && cellId["/"]) {
+              // Found direct cell reference
+              addMatchingCharm(cellId);
+            }
+          }
+        }
+      }
+
+      // Skip the findAllAliasedDocs call since it's causing runtime issues
+      // The flat scan of top-level properties should be sufficient for most cases
```

Comment:
Uh... maybe we should remove this comment and leave a TODO for the recursive version?